### PR TITLE
use a counter for passed tests instead of storing the Result object

### LINKED ISSUE #20137: In testset, use a counter for successes, rather than keeping an array of objects
I'm about to fix the crash in #20107, however, it still runs out of memory, albeit a bit later. This is because the test system keeps a `Test` object for every successful test, which accumulate and take up a large amount of memory. For succesful tests, a simple counter would be sufficient.

### PR DESCRIPTION
**Pull request by KristofferC**
Should fix #20137. cc @Keno

### CODE REVIEW

File: base/test.jl [hunk 1]
```diff
@@ -543,12 +543,15 @@ along with a summary of the test results.
...
     results::Vector
+    n_passed::Int
     anynonpass::Bool
 end
-DefaultTestSet(desc) = DefaultTestSet(desc, [], false)
+DefaultTestSet(desc) = DefaultTestSet(desc, [], 0, false)
 
-# For a passing or broken result, simply store the result
-record(ts::DefaultTestSet, t::Union{Pass,Broken}) = (push!(ts.results, t); t)
+# For a a broken result, simply store the result
```
Comment (tkelman) [inline base/test.jl]: For a broken

**Comment by KristofferC** [comment]
Trvis before latest ci skipped commit at: https://travis-ci.org/JuliaLang/julia/builds/193693108

**Comment by tkelman** [comment]
segfault in valid_type_param in the umfpack test, hm

**Comment by KristofferC** [comment]
how can I build 32 bit to test the actual issue and see it is fixed?

**Comment by tkelman** [comment]
either a vm or there's a docker container i can dig up the link for, or multilib compilers depending on your distro - put these in Make.user and build either in a separate build dir or clean clone.
```
override ARCH = i686
override MARCH = pentium4
```

**Review by kshyatt — APPROVED** [review]

**Comment by tkelman** [comment]
couple weeks old, but here's a docker container with a 32 bit build and all deps already compiled (warning: big download): `docker run -it tkelman/julia32-part1:manual`

**Comment by Keno** [comment]
I can just test this locally. Hold on.

**Comment by Keno** [comment]
Fixes the issue! Great work, thanks!

**Pull request merged**